src/pages/en/guides/migrate-to-astro/from-nextjs.mdx

```diff
@@ -2,72 +2,924 @@
+
+export default class MyDocument extends Document {
+  render() {
+    return (
+      <Html lang="en">
+        <Head>
+          <link rel="icon" href="/favicon.ico" />
+        </Head>
+        <body>
+          <Main />
+          <NextScript />
+        </body>
+      </Html>
+    );
+  }
+}
+```
+
+1. Make a new Astro layout file using only the returned JSX.
+
+2. Replace any React components with `<html>`, `<head>`, `<slot>`, and other HTML standard tags.
+
+
+```astro title="src/layouts/Document.astro"
+<html lang="en">
+	<head>
+	    <link rel="icon" href="/favicon.ico" />
+	</head>
+	<body>
+		<slot/>
+	</body>
+</html>
+```
+
+#### Migrating from Next.js' `/app` directory
+
+Next.js' `app/` directory layout files are created with two files: a `layout.jsx` file to customize the `<html>` and `<body>` contents, and a `body.jsx` file to customize the `<head>` element contents.
+
+```jsx title="app/layout.jsx"
+export default function Layout({ children }) {
+  return (
+    <html lang="en">
+      <body>{children}</body>
+    </html>
+  );
+}
+```
+
+```jsx title="app/head.jsx"
+export default function Head() {
+  return (
+    <>
+      <title>My Page</title>
+    </>
+  );
+}
+```
+
+1. Make a new Astro layout file using only the returned JSX.
+
+2. Replace both these files with a single Astro layout file that contains a page shell (`<html>`, `<head>`, and `<body>` tags) and a `<slot/>` instead of React's `{children}` prop:
+
+```astro title="src/layouts/Layout.astro"
+<html lang="en">
+	<head>
+	    <title>My Page</title>
+	</head>
+	<body>
+		<slot/>
+	</body>
+</html>
+```
+
+### Migrating Pages and Posts
+
+In Next.js, your posts either live in `/pages` or `/app/routeName/page.jsx`.
+
+In Astro, **all your page content must live within `src/`**.
+
+#### React pages
+
+Your existing Next JSX (`.js`) pages will need to be [converted from JSX files to `.astro` pages](#convert-jsx-files-to-astro-files). You cannot use an existing JSX page file in Astro.
+
+These [`.astro` pages](/en/core-concepts/pages/) must be located within `src/pages/` and will have page routes generated automatically based on their file path.
+
+#### Markdown and MDX pages
+
+Astro has built-in support for Markdown and an optional integration for MDX files. You can reuse any existing [Markdown and MDX files](/en/guides/markdown-content/), but they may require some adjustments to their frontmatter, such as adding [Astro's special `layout` frontmatter property](/en/core-concepts/layouts/#markdownmdx-layouts). You will no longer need to manually create pages for each Markdown-generated route. These files can be placed within `src/pages/` to take advantage of automatic file-based routing.
+
+When part of a [content collection](/en/guides/content-collections/), Markdown and MDX files will live in folders within `src/content/` and you will [generate those pages dynamically](/en/guides/content-collections/#generating-pages-from-content-collections).
```

Comment:
I don't think this does enough to explain what a content collection is, especially since there's no equivalent in Next. Maybe something like:

```suggestion
Alternatively, you can use [content collections](/en/guides/content-collections/) to store structured content, such as shop items that all have the same set of properties. When part of a content collection Markdown and MDX files will live in folders within `src/content/`. You will retrieve the content yourself and [generate pages dynamically](/en/guides/content-collections/#generating-pages-from-content-collections).
```